pkg/controller/scyllaclustermigration/sync.go

```diff
@@ -0,0 +1,226 @@
+		ctx,
+		sc,
+		scyllaClusterControllerGVK,
+		scSelector,
+		controllerhelpers.ControlleeManagerGetObjectsFuncs[CT, *corev1.ServiceAccount]{
+			GetControllerUncachedFunc: scmc.scyllaClient.ScyllaV1().ScyllaClusters(sc.Namespace).Get,
+			ListObjectsFunc:           scmc.serviceAccountLister.ServiceAccounts(sc.Namespace).List,
+			PatchObjectFunc:           scmc.kubeClient.CoreV1().ServiceAccounts(sc.Namespace).Patch,
+		},
+	)
+	if err != nil {
+		objectErrs = append(objectErrs, err)
+	}
+
+	err = controllerhelpers.ReleaseObjects[CT, *rbacv1.RoleBinding](
+		ctx,
+		sc,
+		scyllaClusterControllerGVK,
+		scSelector,
+		controllerhelpers.ControlleeManagerGetObjectsFuncs[CT, *rbacv1.RoleBinding]{
+			GetControllerUncachedFunc: scmc.scyllaClient.ScyllaV1().ScyllaClusters(sc.Namespace).Get,
+			ListObjectsFunc:           scmc.roleBindingLister.RoleBindings(sc.Namespace).List,
+			PatchObjectFunc:           scmc.kubeClient.RbacV1().RoleBindings(sc.Namespace).Patch,
+		},
+	)
+	if err != nil {
+		objectErrs = append(objectErrs, err)
+	}
+
+	err = controllerhelpers.ReleaseObjects[CT, *policyv1.PodDisruptionBudget](
+		ctx,
+		sc,
+		scyllaClusterControllerGVK,
+		scSelector,
+		controllerhelpers.ControlleeManagerGetObjectsFuncs[CT, *policyv1.PodDisruptionBudget]{
+			GetControllerUncachedFunc: scmc.scyllaClient.ScyllaV1().ScyllaClusters(sc.Namespace).Get,
+			ListObjectsFunc:           scmc.pdbLister.PodDisruptionBudgets(sc.Namespace).List,
+			PatchObjectFunc:           scmc.kubeClient.PolicyV1().PodDisruptionBudgets(sc.Namespace).Patch,
+		},
+	)
+	if err != nil {
+		objectErrs = append(objectErrs, err)
+	}
+
+	err = controllerhelpers.ReleaseObjects[CT, *networkingv1.Ingress](
+		ctx,
+		sc,
+		scyllaClusterControllerGVK,
+		scSelector,
+		controllerhelpers.ControlleeManagerGetObjectsFuncs[CT, *networkingv1.Ingress]{
+			GetControllerUncachedFunc: scmc.scyllaClient.ScyllaV1().ScyllaClusters(sc.Namespace).Get,
+			ListObjectsFunc:           scmc.ingressLister.Ingresses(sc.Namespace).List,
+			PatchObjectFunc:           scmc.kubeClient.NetworkingV1().Ingresses(sc.Namespace).Patch,
+		},
+	)
+	if err != nil {
+		objectErrs = append(objectErrs, err)
+	}
+
+	err = controllerhelpers.ReleaseObjects[CT, *batchv1.Job](
+		ctx,
+		sc,
+		scyllaClusterControllerGVK,
+		scSelector,
+		controllerhelpers.ControlleeManagerGetObjectsFuncs[CT, *batchv1.Job]{
+			GetControllerUncachedFunc: scmc.scyllaClient.ScyllaV1().ScyllaClusters(sc.Namespace).Get,
+			ListObjectsFunc:           scmc.jobLister.Jobs(sc.Namespace).List,
+			PatchObjectFunc:           scmc.kubeClient.BatchV1().Jobs(sc.Namespace).Patch,
+		},
+	)
+	if err != nil {
+		objectErrs = append(objectErrs, err)
+	}
+
+	objectErr := utilerrors.NewAggregate(objectErrs)
+	if objectErr != nil {
+		return objectErr
+	}
+
+	sdc, err := scmc.scyllaDBDatacenterLister.ScyllaDBDatacenters(namespace).Get(name)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
```

Comment:
this needs to check the controller ref before counting it into the status - generally the `controllerhelpers.GetObjects` pattern and propagate it further

Reply:
done